src/components/Navbar/components/ProfileModal/index.tsx

```diff
@@ -0,0 +1,150 @@
+// Components
+import { ConfigProvider, Modal } from "antd";
+
+// Assets
+import avatar1 from "@/assets/avatar1@3x.png";
+
+// Utils
+import useAuth from "@/hooks/useAuth";
+import { getUserAvatar } from "@/utils/userUtils";
+import { useEffect, useState } from "react";
+
+// Constants - types
+import { UserInfoType } from "@/types/userTypes";
+import AvatarModal from "./AvatarModal";
+
+export default function ProfileModal({
+	isModalOpen,
+	setIsModalOpen,
+}: {
+	isModalOpen: boolean;
+	setIsModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}) {
+	const { userInfo, logout, updateUserData } = useAuth();
+
+	const [school, setSchool] = useState<string>("");
+	const [grade, setGrade] = useState<string>("");
+	const [isLoading, setIsLoading] = useState<boolean>(false);
+	const [isAvatarModalOpen, setIsAvatarModalOpen] = useState<boolean>(false);
+
+	useEffect(() => {
+		if (userInfo) {
+			setSchool(userInfo.school as string);
+			setGrade(userInfo.grade as string);
+		}
+	}, [userInfo]);
+
+	const handleSave = async () => {
+		setIsLoading(true);
+		await updateUserData({
+			school,
+			grade,
+		});
+		setIsLoading(false);
+		setIsModalOpen(false);
+	};
+
+	const handleLogout = async () => {
+		await logout();
+		setIsModalOpen(false);
+	};
+
+	return (
+		<ConfigProvider
+			theme={{
+				components: {
+					Modal: {
+						contentBg: "#fcf1e6",
+					},
+				},
+			}}
+		>
+			<Modal
+				open={isModalOpen}
+				onCancel={() => setIsModalOpen(false)}
+				width={960}
+				footer={null}
+			>
+				<div className="flex flex-col items-center gap-8 my-2">
+					<div className="text-xl sm:text-2xl font-semibold">
+						Profile
+					</div>
+					<div className="flex flex-col items-center gap-1">
+						<AvatarModal
+							isModalOpen={isAvatarModalOpen}
+							setIsModalOpen={setIsAvatarModalOpen}
+						/>
+						<img
+							src={
+								userInfo
+									? getUserAvatar(userInfo as UserInfoType)
+									: avatar1
+							}
+							className="w-16 h-16 sm:w-20 sm:h-20 rounded-full"
+						/>
+						<div
+							className="underline cursor-pointer"
+							onClick={() => setIsAvatarModalOpen(true)}
+						>
+							Change Avatar
+						</div>
+					</div>
+					<div className="sm:flex flex-row gap-6 md:gap-12 w-11/12 md:w-5/6 justify-between">
+						<div className="flex flex-col flex-1 gap-2">
+							<div className="text-base md:text-lg font-medium">
+								Manage your account
+							</div>
+							<div className="flex flex-col gap-4">
+								<input
+									type="text"
+									id="school"
+									className="block bg-transparent w-full border border-solid border-gray-400 rounded-lg focus:border-red-800  focus:ring-red-800 transition-all sm:text-base px-4 py-2 text-ellipsis"
+									placeholder="School"
+									required
+									value={school}
+									onChange={(e) => setSchool(e.target.value)}
+								/>
+								<input
```

Comment:
I think would be nice to disable the button when the input.event.values are empty

Reply:
Maybe show the school and grade value as placeholder for the inputs

Reply:
maybe we can put an edit button to enable edit for the manage your account part so that the user can see the info and able to edit it

Reply:
will add the disable button. as for the placeholder, if the data for school and grade is not empty, it will serve as a default value and will show in the input value. hence, that won't be a problem